ext/lvgl/lv_conf.h.in

```diff
@@ -0,0 +1,757 @@
+/**
+ * @file lv_conf.h
+ * Configuration file for v7.11.0
+ */
+
```

Comment:
```suggestion
#if __has_include(<lv_conf_local.h>)
#include <lv_conf_local.h>
#endif
```

Perhaps something like this to overwrite this file by the user. I don't really want to push all the settings through lbuild that seems stupid. We've done something like this for FreeRTOS too.

Reply:
... but at the end of the `lv_conf.h.in` file so that the user can overwrite the defines.

Reply:
How does this work for FreeRTOS?

```
In file included from modm/ext/lvgl/lvgl/src/lv_misc/../lv_conf_internal.h:38,
                 from modm/ext/lvgl/lvgl/src/lv_misc/lv_log.h:16,
                 from modm/ext/lvgl/lvgl/lvgl.h:25,
                 from modm/ext/lvgl/lvgl/lv_modm_logging.cpp:11:
modm/ext/lvgl/lv_conf.h:759:12: fatal error: lv_conf_local.h: No such file or directory
  759 | #  include <lv_conf_local.h>
      |            ^~~~~~~~~~~~~~~~~
```

File `lv_conf_local.h` exists in project top level directory and `__has_include(<lv_conf_local.h>)` is commented for debugging.

Reply:
Hm… I tested this with `tusb_config_local.h` for my submodule update branch. It's just a file with an `#error "foo"` and it gets found correctly.

I'll check how the freertos one does it.

Reply:
Hm, it's also called `FreeRTOSConfigLocal.h`… #NamingThingsConsistently…

Reply:
This also works for me:
```
In file included from modm/ext/freertos/inc/freertos/FreeRTOSConfig.h:150,
                 from modm/ext/freertos/inc/freertos/FreeRTOS.h:57,
                 from modm/src/modm/processing/rtos/scheduler.hpp:18,
                 from modm/src/modm/processing/rtos.hpp:22,
                 from main.cpp:15:
./FreeRTOSConfigLocal.h:1:2: error: #error "foo"
    1 | #error "foo"
      |  ^~~~~
```

Reply:
I'm using SCons, perhaps the other build systems are missing an include path…

Reply:
Ah… You need to do a `scons -c` first, because it doesn't detect the dependency correctly.

Reply:
Nope, I get the same error for FreeRTOS+TCP, while it works fine for FreeRTOS.

```
In file included from modm/ext/freertos_plus_tcp/include/FreeRTOS_IP.h:37,
                 from modm/ext/freertos_plus_tcp/tools/tcp_utilities/tcp_dump_packets.c:43:
modm/ext/freertos_plus_tcp/include/FreeRTOSIPConfig.h:310:10: fatal error: FreeRTOSIPConfigLocal.h: No such file or directory
  310 | #include <FreeRTOSIPConfigLocal.h>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
```

Reply:
```
arm-none-eabi-gcc -o /home/user/modm/build/nucleo_l452re/lvgl/scons-release/modm/ext/lvgl/lvgl/src/lv_font/lv_font_montserrat_36.o -c -std=gnu11 -Wbad-function-cast -Wimplicit -Wredundant-decls -Wstrict-prototypes -fdata-sections -ffile-prefix-map=/opt/rleh/gcc-arm-none-eabi-10-2020-q4-major=. -ffile-prefix-map=/home/user/modm/examples/nucleo_l452re/lvgl=. -ffunction-sections -finline-limit=10000 -fshort-wchar -fsingle-precision-constant -funsigned-bitfields -funsigned-char -fwrapv -g3 -gdwarf-3 -W -Wall -Wdouble-promotion -Wduplicated-cond -Werror=format -Werror=maybe-uninitialized -Werror=overflow -Werror=sign-compare -Wextra -Wlogical-op -Wno-redundant-decls -Wpointer-arith -Wundef -Os -mcpu=cortex-m4 -mfloat-abi=hard -mfpu=fpv4-sp-d16 -mthumb -Imodm/ext -Imodm/ext/cmsis/core -Imodm/ext/cmsis/device -Imodm/ext/crashcatcher/include -Imodm/ext/lvgl -Imodm/src -Imodm modm/ext/lvgl/lvgl/src/lv_font/lv_font_montserrat_36.c

[...]

/opt/rleh/gcc-arm-none-eabi-10-2020-q4-major/bin/../lib/gcc/arm-none-eabi/10.2.1/../../../../arm-none-eabi/bin/ld: /home/user/modm/build/nucleo_l452re/lvgl/scons-release/main.o: in function `main':
./main.cpp:197: undefined reference to `lv_font_montserrat_36'
collect2: error: ld returned 1 exit status
```

The include path `-I.` is missing for `lv_font_montserrat_36.c` (but it is there for `main.cpp`).

Reply:
Yup, I found the issue a missing `abspath`. The Makefile generator also had the `-I.` missing completely… CMake did have the issue.

Reply:
Fixed by @salkinium in 9a0bc1b